src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,15 @@
+export const useCopyToClipboard = () => {
+  const copy = async (value: string) => {
```

Comment:
Also, this doesn't need to be a hook, as it doesn't use any of React's life-cycle's.

It could just be a function on `src/utils` such as `copyTextToClipboard.ts`

Reply:
Hence it can be simplifed to:

```jsx
export const copyTextToClipboard = async (value: string) => {
  if (typeof navigator !== 'object' || !navigator.clipboard) {
    return false;
  }

  return navigator.clipboard.writeText(value)
    .then(() => true)
    .catch(() => false);
}

// Then in the usages
const result = copyTextToClipboard(textToCopy);

setCopied(await result);
```